frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -87,22 +135,39 @@ const GitSection: React.FC<GitSectionProps> = ({
       }
 
       setFieldValue('git.isUrlValidating', true);
-      setValidated(ValidatedOptions.default);
 
-      const gitType = gitTypeTouched ? values.git.type : detectGitType(url);
+      const gitService = getGitService(
+        url,
+        values.git.type,
+        ref,
+        dir,
+        values.git.secretResource,
+        values.devfile?.devfilePath,
+        values.docker.dockerfilePath,
+      );
+
+      const importStrategyData = await detectImportStrategies(url, gitService, values.git.type);
+
+      const {
+        loaded,
+        loadError,
+        repositoryStatus,
+        strategies: importStrategies,
+        gitProvider,
+      } = importStrategyData;
+
+      const gitType = gitTypeTouched ? values.git.type : gitProvider;
       const gitRepoName = detectGitRepoName(url);
-      const showGitType = gitType === GitTypes.unsure || (gitTypeTouched && !isSubmitting);
+      const showGitType = gitProvider === GitProvider.UNSURE || (gitTypeTouched && !isSubmitting);
 
       setFieldValue('git.type', gitType);
       setFieldValue('git.showGitType', showGitType);
```

Comment:
Git type detection and selection needs to happen before you create a git service. This is why the warning pops up without validating anything.